### PR TITLE
Let Autocomplete use the same show option as the Select component

### PR DESCRIPTION
**Pull request by clemens-msupply**
Part of #2329

For the HIV contact tracing I tried to keep the original codes which are stored in original data, e.g. "C" for a "Children" relationship.

This PR replaces the existing `values` option for the Autocomplete control with a `show` option similar to the `show` option of the Select control. I thought it would be nice to have the same option and to be able to customize how thing are displayed and how they are stored.

However, there is one problem. The contact tracing table column now shows "C" instead of "Children". Had a chat with Richard, and he was fine with it. If this is a problem we can still update the UI schema to use the display options...

### CODE REVIEW

File: client/packages/programs/src/JsonForms/common/components/Autocomplete.tsx [hunk 1]
```diff
@@ -9,21 +9,35 @@ import { z } from 'zod';
...
 import { DefaultFormRowSx, FORM_LABEL_WIDTH } from '../styleConstants';
 
-type Options = {
-  freeText?: boolean;
-  values?: string[];
-};
-const Options: z.ZodType<Options | undefined> = z
+const Options = z
   .object({
     freeText: z.boolean().optional(),
```
Comment (clemens-msupply) [inline client/packages/programs/src/JsonForms/common/components/Autocomplete.tsx:14]: freeText is currently not used and I haven't tested it. Hopefully it is still working though...

Comment (roxy-dao) [reply client/packages/programs/src/JsonForms/common/components/Autocomplete.tsx:14]: Doesn't seem like it's working anymore...

Comment (roxy-dao) [reply client/packages/programs/src/JsonForms/common/components/Autocomplete.tsx:14]: Doesn't seem like it's working anymore...

File: client/packages/programs/src/JsonForms/common/components/Autocomplete.tsx [hunk 2]
```diff
@@ -50,38 +76,33 @@ const UIComponent = (props: ControlProps) => {
...
   };
 
   const onChange = (
     _event: React.SyntheticEvent,
-    value: DisplayOption | null | string
+    option: DisplayOption | null
   ) => {
     let s = '';
-    if (typeof value === 'string') {
+    if (typeof option === 'string') {
```
Comment (roxy-dao) [inline client/packages/programs/src/JsonForms/common/components/Autocomplete.tsx:88]: This doesn't trigger at all...

**Review by roxy-dao — CHANGES_REQUESTED** [review]
Thanks Clemens~

Everything looks great, but the freeText doesn't seem to be working anymore...

File: client/packages/programs/src/JsonForms/common/components/Autocomplete.tsx (hunk 2 above)
Comment (clemens-msupply) [reply client/packages/programs/src/JsonForms/common/components/Autocomplete.tsx]: that was part of why free text import was not working...

**Comment by clemens-msupply** [comment]
Thanks I made some changes to fix free text input. Think MUI update fixed the underlying problem that I faced previously, i.e. code is much simpler now.

**Review by roxy-dao — APPROVED** [review]
Great~ Changes look much neater and works like a charm (:

**Pull request merged**